pilot/pkg/serviceregistry/kube/controller/network_test.go

```diff
@@ -237,6 +237,53 @@ func TestSyncAllWorkloadsFromAmbient(t *testing.T) {
 
 	s := newAmbientTestServer(t, testC, "")
 
+	namespaceNotifyCh := make(chan string, 5)
+	defer close(namespaceNotifyCh)
+	s.controller.namespaces.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+		namespaceNotifyCh <- o.GetName()
+	}))
+
+	waitNamespaceNotify := func(namespace string) {
+		retry.UntilSuccessOrFail(t, func() error {
+			select {
+			case notifiedNamespace := <-namespaceNotifyCh:
+				if notifiedNamespace != namespace {
+					return fmt.Errorf("no namespace notify")
+				}
+			case <-time.After(5 * time.Second):
```

Comment:
This is not needed

Reply:
Found an issue with the retry logic: if we remove this and do not send things into the channel, the test will wait forever instead of 5s timeout. I'm going to send a fix.